rest-service/manager_rest/plugins_update/manager.py

```diff
@@ -0,0 +1,245 @@
+#########
+# Copyright (c) 2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+
+import os
+import uuid
+
+from flask import current_app
+
+from cloudify.models_states import ExecutionState
+
+from dsl_parser.constants import (PLUGIN_NAME_KEY,
+                                  WORKFLOW_PLUGINS_TO_INSTALL,
+                                  DEPLOYMENT_PLUGINS_TO_INSTALL)
+
+from manager_rest import config, utils
+from manager_rest.storage import get_storage_manager, models
+from manager_rest.resource_manager import get_resource_manager
+from manager_rest.plugin_update.constants import (STATES,
+                                                  ACTIVE_STATES)
+from manager_rest.manager_exceptions import (ConflictError,
+                                             PluginsUpdateError,
+                                             IllegalActionError)
+from manager_rest.constants import FILE_SERVER_BLUEPRINTS_FOLDER
+
+
+class PluginsUpdateManager(object):
+    def __init__(self, sm):
+        self.sm = sm
+        self.rm = get_resource_manager(self.sm)
+
+    def validate_no_active_updates_per_blueprint(self,
+                                                 blueprint_id,
+                                                 force=False):
+        active_updates = self.list_plugins_updates(
+            filters={'blueprint_id': blueprint_id,
+                     'state': ACTIVE_STATES}).items
+        if not active_updates:
+            return
+
+        if not force:
+            raise ConflictError(
+                'There are plugins updates still active, update IDs: '
+                '{0}'.format(', '.join([u.id for u in active_updates])))
```

Comment:
remember that `str.join` takes iterables, and a generator expression is fine, so this can just be 
```python
', '.join(u.id for u in active_updates)
```
(without the `[ ]`)

Reply:
Woops